### ISSUE TITLE
Improvement: use typing information to prevent "Add override if that's the reason"

### ISSUE THREAD

**Issue by jrudolph** [HairyFotr/linter#8]
```
[warn] /home/johannes/git/telfish/spray/spray/spray-can/src/main/scala/spray/can/server/PipeliningLimiter.scala:33: Parameter context is not used in method apply. (Add override if that's the reason)
[warn]       def apply(context: PipelineContext, commandPL: CPL, eventPL: EPL): Pipelines =
[warn]           ^
```

You could use typing information instead of relying on the modifiers to find out if a method is an implementation of an abstract method.

On the other hand, it's arguable if you should always use override in these cases but there are a lot of cases where adding `override` just adds noise (like SAMs). Maybe make it configurable if you want all method implementations to use override and other wise don't warn when you can detect that a method is an implementation.
